WixSharp/Services/WixService.cs

```diff
@@ -0,0 +1,311 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using WixSharp.Infrastructure;
+
+namespace WixSharp
+{
+    public abstract class WixService
+    {
+        private static IRequestExecutionPolicy _GlobalExecutionPolicy = new DefaultRequestExecutionPolicy();
+
+        private IRequestExecutionPolicy _ExecutionPolicy;
+
+        private static JsonSerializer _Serializer = Serializer.JsonSerializer;
+
+        private static HttpClient _Client { get; } = new HttpClient();
+
+        protected Uri _ShopUri { get; set; }
+
+        protected string _AccessToken { get; set; }
+
+        /// <summary>
+        /// Creates a new instance of <see cref="WixService" />.
+        /// </summary>
+        /// <param name="myWixUrl">The shop's *.myWix.com URL.</param>
+        /// <param name="shopAccessToken">An API access token for the shop.</param>
+        protected WixService(string shopAccessToken, string myWixUrl)
+        {
+            _ShopUri = BuildShopUri(false, myWixUrl);
+            _AccessToken = shopAccessToken;
+
+            // If there's a global execution policy it should be set as this instance's policy.
+            // User can override it with instance-specific execution policy.
+            _ExecutionPolicy = _GlobalExecutionPolicy ?? new DefaultRequestExecutionPolicy();
+        }
+
+        /// <summary>
+        /// Attempts to build a shop API <see cref="Uri"/> for the given shop. Will throw a <see cref="WixException"/> if the URL cannot be formatted.
+        /// </summary>
+        /// <param name="myWixUrl">The shop's *.myWix.com URL.</param>
+        /// <exception cref="WixException">Thrown if the given URL cannot be converted into a well-formed URI.</exception>
+        /// <returns>The shop's API <see cref="Uri"/>.</returns>
+        public static Uri BuildShopUri(bool withAdminPath, string myWixUrl)
+        {
+            if (string.IsNullOrEmpty(myWixUrl)) myWixUrl = "www.wixapis.com";
+
+            if (Uri.IsWellFormedUriString(myWixUrl, UriKind.Absolute) == false)
+            {
+                //Wix typically returns the shop URL without a scheme. If the user is storing that as-is, the uri will not be well formed.
```

Comment:
We need to handle wix url generation. The URL must have either `http` or `https`